service/storage_service.cc

```diff
@@ -824,7 +827,7 @@ storage_service::get_range_to_address_map(const sstring& keyspace,
     auto& rs = _db.local().find_keyspace(ks).get_replication_strategy();
     auto ranges = get_all_ranges(sorted_tokens);
     for (auto& r : ranges) {
-        res[r] = rs.get_natural_endpoints(r.end() ? r.end()->value() : dht::maximum_token());
+        res[r] = rs.get_natural_endpoints(r.end() ? r.end()->value() : dht::maximum_token(), locator::can_yield::yes);
```

Comment:
Ok, I see why you create some threads. The low-level APIs infected the high-level APIs. That's not good.

Reply:
I think it's better to return a future and construct the thread here. This way if we later have a non-threaded implementation, we don't have to unravel the whole stack.